src/libs/actions/Report.ts

```diff
@@ -1095,12 +1043,6 @@ function handleReportChanged(report: OnyxEntry<Report>) {
             conciergeChatReportID = report.reportID;
         }
     }
-
-    // A report can be missing a name if a comment is received via pusher event and the report does not yet exist in Onyx (eg. a new DM created with the logged in person)
-    // In this case, we call reconnect so that we can fetch the report data without marking it as read
-    if (report.reportID && report.reportName === undefined) {
-        reconnect(report.reportID);
-    }
```

Comment:
TODO: Need to make sure this isn't needed. If reliable updates works properly, I don't think we should miss a report, but 🤷

Reply:
This code is very very very very Old (~2020), and it is very likely that things were much more unreliable back then. I think this came from this PR:

https://github.com/Expensify/App/pull/339/files

And have been modified a few times after like for example here: https://github.com/Expensify/App/pull/1075/files#r548053786

I think can be tested by:

1. `accountA` is offline and has never chatted with `accountB` before
2. `accountB` creates a DM with `accountA` and sends a message. `accountA` misses the push event
3. `accountA` goes online
4. `accountB` sends another message

That way, when we push the last message sent by `accountB` in step 4, `accountA` could have been missing the report, but that is not the case because `accountA` got the report in step 3 because of "reliable updates".

Reply:
Agree with you @aldo-expensify that the app is more reliable. I think this code is _basically_ here for the case you have laid out i.e. it was possible to get a pusher update for a report that you didn't have locally yet. And in that case, we would fetch it.

cc @tgolen to get more eyes on this one just in case he can think of something we are not seeing.